### ISSUE TITLE
[C#] Rainbow variables does not work as intended

### ISSUE THREAD

**Issue by jansviland** [izhangzhihao/intellij-rainbow-brackets#2626]
1. First issue is that fromTime, toTime, body, response and deserialize is more or less the same color? If there is a difference, I can not tell.
2. Second issue is that the body request does not work as intented, it is two different colors. When it should be one.  

![bilde](https://user-images.githubusercontent.com/114911422/235885619-329041ff-420d-4c4b-afee-f1d121152ee6.png)

Here is my config

![bilde](https://user-images.githubusercontent.com/114911422/235886074-aa1560b9-2b15-49f9-abbb-991d353c3a28.png)

## Please check

- [X] I already support this project.
- [X] I already paid for this plugin.
- [ ] I'm on a trial of this plugin.
- [ ] This issue/feature request is not reported before.

This has been reported before, and fixed, but then it seems like the bug is back again?

## Your programming languages

C#

```
JetBrains Rider 2023.1.1
Build #RD-231.8770.54, built on April 25, 2023
Licensed to ....
You have a perpetual fallback license for this version.
Subscription is active until May 24, 2024.
Runtime version: 17.0.6+10-b829.9 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10.0
.NET 7.0.2 (Server GC)
GC: G1 Young Generation, G1 Old Generation
Memory: 1500M
Cores: 12
Registry:
    ide.lookup.shrink=true
    ide.experimental.ui=true
    vcs.empty.toolwindow.show=false
    rdclient.asyncActions=false
    ide.new.project.model.index.case.sensitivity=true
    database.show.search.tab=false

Non-Bundled Plugins:
    com.intellij.javafx (1.0.4)
    Batch Scripts Support (1.0.13)
    com.intellij.resharper.HeapAllocationsViewer (2023.1.0)
    IdeaVIM (2.2.0)
    org.intellij.plugins.hcl (231.8109.91)
    com.github.copilot (1.2.5.2507)
    avalonia-rider (1.0.3)
    com.markskelton.one-dark-theme (5.7.4)
    fr.socolin.application-insights-debug-log-viewer (1.8.0)
    krasa.CpuUsageIndicator (1.15)
    com.intellij.resharper.StructuredLogging (2023.1.0.296)
    com.samdark.intellij-visual-studio-code-dark-plus (2.6)
    me.seclerp.rider.plugins.efcore (231.1.2)
    com.intellij.resharper.azure (3.50.0.1595-2023.1)
    izhangzhihao.rainbow.brackets (2023.2.7)

```

**Comment by jansviland** [comment]
I submitted a very similar issue here: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2612

And it was fixed, but I guess it's back again?

**Comment by izhangzhihao** [comment]
please provide your code sample, I will try find why this happens.

**Comment by jansviland** [comment]
I was able to reproduce it with this code, just a simple console app: 

```C#
// See https://aka.ms/new-console-template for more information

Console.WriteLine("Hello, World!");

// arrange
const string fromTime = $"2018-01-01";
const string toTime = $"2023-01-01";

var body = new TimeSeriesFilter()
{
    Selections = new List<TimeSeriesFilterSelection>()
    {
        new()
        {
            StationIds = new[] { "456" },
            ComponentIds = new[] { "123" },
            QualityLevel = new[] { QualityLevelEnum.BestAvailable }
        }
    },
    Period = PeriodTypeEnum.Hour
};

var request = CreatePostRequest($"/portal/timeseries/{fromTime}/{toTime}/names", body);

object CreatePostRequest(string s, TimeSeriesFilter timeSeriesFilter)
{
    throw new NotImplementedException();
}

public enum QualityLevelEnum
{
    BestAvailable,
    RawData,
    ValidData
}

public enum PeriodTypeEnum
{
    Minute,
    TenMinutes,
    Hour,
    Day,
    Month,
    Year,
    Unknown
}

public class TimeSeriesFilter
{
    public List<TimeSeriesFilterSelection> Selections { get; init; } = new();
    public PeriodTypeEnum Period { get; init; }
}

public class TimeSeriesFilterSelection
{
    public string[] StationIds { get; init; }
    public string[] ComponentIds { get; init; }
    public QualityLevelEnum[] QualityLevel { get; init; }
}

```

**Comment by jansviland** [comment]
Here is how it looks

![bilde](https://user-images.githubusercontent.com/114911422/235922552-40f40d87-eb63-43fe-96af-848b1e98898f.png)

**Comment by jansviland** [comment]
also, if I add a couple of variables here, you can see that the colors are almost or identical. 

![bilde](https://user-images.githubusercontent.com/114911422/235922973-c2b36332-f366-491a-b1e5-fc6d3081bf0d.png)

**Comment by izhangzhihao** [comment]
Please try this build snapshot: 
[intellij-rainbow-brackets-2023.2.7.zip](https://github.com/izhangzhihao/intellij-rainbow-brackets/files/11392283/intellij-rainbow-brackets-2023.2.7.zip)

**Comment by izhangzhihao** [comment]
I enhanced the random color generator, and now colors have a higher degree of differentiation from each other(Could be a new config in the future), please have a try and give me feedback, thanks.

[intellij-rainbow-brackets-2023.2.7.zip](https://github.com/izhangzhihao/intellij-rainbow-brackets/files/11412162/intellij-rainbow-brackets-2023.2.7.zip)

**Comment by jansviland** [comment]
<img width="869" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/114911422/39a04e88-404d-4672-a722-b3231078fd02">

Looks better now 👍

**Comment by jansviland** [comment]
I was unfortunately able to reproduce the issue again: 

<img width="694" alt="Screenshot 2023-05-22 at 16 19 17" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/114911422/c7f276f4-b990-4c93-a0ee-275c7fa60491">

First has the wrong colour. 

And the colours selected for first, second and remainder are very similar.

**Comment by jansviland** [comment]
<img width="692" alt="Screenshot 2023-05-23 at 10 43 12" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/114911422/ba33e7cd-a284-4f16-9805-7aae44bb71ed">

Another strange behaviour I have discovered recently, sometimes the parameter, in this case just the customers string url, is treated as a variable. And also gets another colour. I does not always happen. Like here it happens the first time, but not the second time, even when it's the same method.

**Comment by jansviland** [comment]
<img width="920" alt="Screenshot 2023-05-23 at 10 50 41" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/114911422/2005feb1-a423-4fbf-88f1-28ad80531d55">

Another example, here the int value 1, gets a colour. This should not happen

**Comment by izhangzhihao** [comment]
please post a full example code, so I could reproduce this case easily

**Comment by jansviland** [comment]
Sure, you can find the examples above here: 

https://github.com/jansviland/testing-workshop-ndcoslo-2023/blob/bug/1.UnitTesting/5.MutationTesting/src/AwesomeCalculator/Calculator.cs

https://github.com/jansviland/testing-workshop-ndcoslo-2023/blob/bug/2.IntegrationTesting/1.ApiTesting/tests/Customers.Api.Tests.Integration/CustomerEndpointsTests.cs

**Comment by izhangzhihao** [comment]
It seems fixed in the latest Rider EAP(version 2023.2 EAP 6).

<img width="620" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/c97e5cb9-0df2-46a8-9ce4-98d82147fea1">

<img width="686" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/8f724a56-4519-4255-b106-42e1303f86c6">


```
JetBrains Rider 2023.2 EAP 6
Build #RD-232.8296.16, built on June 26, 2023
Licensed to JetBrains Rider EAP user: Zhang Zhihao
Expiration date: July 26, 2023
Runtime version: 17.0.7+7-b1000.2 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6.3
.NET Core v7.0.7 x64 (Server GC)
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Metal Rendering is ON
Registry:
    ide.plugins.snapshot.on.unload.fail=true
    ide.experimental.ui=true
    database.show.search.tab=false
    vcs.empty.toolwindow.show=false
    ide.new.project.model.index.case.sensitivity=true

Non-Bundled Plugins:
    PsiViewer (232.2)
    PythonCore (232.8296.17)
    izhangzhihao.rainbow.brackets (2023.2.11)
```

**Comment by izhangzhihao** [comment]
Please rise a new issue if there are still some cases that don't work well.
